src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpTagHelperRunnerInitializationVisitor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.Generator.Compiler.CSharp
+{
+    public class CSharpTagHelperRunnerInitializationVisitor : CodeVisitor<CSharpCodeWriter>
+    {
+        private readonly GeneratedTagHelperContext _tagHelperContext;
+        private bool _foundTagHelpers;
+
+        public CSharpTagHelperRunnerInitializationVisitor([NotNull] CSharpCodeWriter writer,
+                                                          [NotNull] CodeBuilderContext context)
+            : base(writer, context)
+        {
+            _tagHelperContext = Context.Host.GeneratedClassContext.GeneratedTagHelperContext;
+        }
+
+        protected override void Visit(TagHelperChunk chunk)
+        {
+            if (!_foundTagHelpers && !Context.Host.DesignTimeMode)
+            {
+                _foundTagHelpers = true;
+
+                Writer.WriteStartAssignment(CSharpTagHelperCodeRenderer.RunnerVariableName)
+                      .Write(CSharpTagHelperCodeRenderer.RunnerVariableName)
+                      .Write(" ?? ")
+                      .WriteStartNewObject(_tagHelperContext.RunnerTypeName)
+                      .Write(_tagHelperContext.HtmlEncoderPropertyName)
+                      .WriteEndMethodInvocation();
```

Comment:
FYI (no need to change anything in this PR) it's much easier and helps future developers if you just let VS do its thing.  in this case that would look more like
``` c#
Writer
    .WriteStartAssignment(CSharpTagHelperCodeRenderer.RunnerVariableName)
    .Write(CSharpTagHelperCodeRenderer.RunnerVariableName)
    ...;
```